qiskit_experiments/framework/base_experiment.py

```diff
@@ -265,31 +265,92 @@ def _finalize(self):
+        max_circuits_option = getattr(self.experiment_options,
+                                      "max_circuits",
+                                       None)
         max_circuits_backend = self._backend_data.max_circuits
+
         if max_circuits_option and max_circuits_backend:
-            max_circuits = min(max_circuits_option, max_circuits_backend)
+            return min(max_circuits_option, max_circuits_backend)
         elif max_circuits_option:
-            max_circuits = max_circuits_option
+            return max_circuits_option
         else:
-            max_circuits = max_circuits_backend
+            return max_circuits_backend
 
-        # Run experiment jobs
-        if max_circuits and len(circuits) > max_circuits:
-            # Split jobs for backends that have a maximum job size
-            job_circuits = [
-                circuits[i : i + max_circuits] for i in range(0, len(circuits), max_circuits)
-            ]
+    def job_info(
+            self,
+            circuits: List[QuantumCircuit],
+            backend: Backend =None):
+        """
+        Get information about job distribution for the experiment on a specific
+        backend.
+
+        Args:
+            circuits (List[QuantumCircuit]): A list of QuantumCircuit objects.
+            backend (Backend): The backend for which to get job distribution 
+            information.
+    
+        Returns:
+            dict: A dictionary containing information about job distribution.
+                - "Total Number of circuits in the experiment": Total number of
+                circuits in the experiment.
+                - "Maximum Number of circuits": Maximum Number of circuits in 
+                one Job
+                - "Total Number of jobs": Number of jobs needed for
+                distribution. 
+
+        Raises:
+            QiskitError: if backend is not specified.
+
+        """
+        max_circuits = self._max_circuits(backend)
+        total_circuits = len(circuits)
+        
+        if total_circuits <= max_circuits:
+            return {
+                "Total Number of circuits in the experiment": total_circuits,
+                "Maximum Number of circuits": max_circuits,
+                "Total Number of jobs": 1
+            }
         else:
-            # Run as single job
+            num_jobs = (total_circuits + max_circuits - 1) // max_circuits
+            return {
+                "Total Number of circuits in the experiment": total_circuits,
+                "Maximum Number of circuits": max_circuits,
+                "Total Number of jobs": num_jobs
+            }
+
+    def _run_jobs(
+            self, 
+            circuits: List[QuantumCircuit], 
+            backend: Backend =None,
+            **run_options) -> List[Job]:
+        """Run circuits on backend as 1 or more jobs."""
+        max_circuits = self._max_circuits(backend)
+        
+        if len(circuits) <= max_circuits:
```

Comment:
This needs to handle the case when `_max_circuits()` returns `None`.

Reply:
I am thinking to solve with this logic

```python
if max_circuits and (len(circuits) <= max_circuits):
            job_circuits = [circuits]
elif max_circuits and (len(circuits) > max_circuits):
            job_circuits = [
                circuits[i:i + max_circuits] for i in range(0, len(circuits), max_circuits)
            ]
```